uitest/Common/DevHomeApplication.cs

```diff
@@ -0,0 +1,95 @@
+﻿// Copyright (c) Microsoft Corporation and Contributors
+// Licensed under the MIT license.
+
+using DevHome.UITest.Configurations;
+using DevHome.UITest.Pages;
+using Microsoft.Extensions.Configuration;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Appium.Windows;
+using OpenQA.Selenium.Support.Extensions;
+
+namespace DevHome.UITest.Common;
+
+/// <summary>
+/// Singleton class representing a Dev Home application instance
+/// </summary>
+public sealed class DevHomeApplication
+{
+    private static readonly Lazy<DevHomeApplication> _instance = new (() => new ());
+    private DevHomeSession _devHomeSession;
+
+    /// <summary>
+    /// Gets the singleton Dev Home application instance
+    /// </summary>
+    public static DevHomeApplication Instance => _instance.Value;
+
+    /// <summary>
+    /// Gets the application settings configuration
+    /// </summary>
+    /// <remarks>Content populated by the appsettings JSON files</remarks>
+    public AppConfiguration Configuration { get; private set; }
+
+    private WindowsElement DashboardNavigationItem => _devHomeSession.Driver.FindElementByAccessibilityId("DevHome.Dashboard");
+
+    private WindowsElement MachineConfigurationNavigationItem => _devHomeSession.Driver.FindElementByAccessibilityId("DevHome.SetupFlow");
+
+    private DevHomeApplication()
+    {
+    }
+
+    /// <summary>
+    /// Initialize the singleton instance
+    /// </summary>
+    /// <param name="appSettingsMode">Application settings mode (local, canary, etc ...)</param>
+    public void Initialize(string appSettingsMode)
```

Comment:
Where does this value come from? Do we need any validation that it's one of the two allowed values?

Reply:
This value is set from the Test.runsettings, which can also be overwritten from the azure pipeline yaml or command line. This parameter defines which appsettings JSON to load last (overwriting previous settings). If we pass an invalid argument (hence file not found) the attempt to apply the settings is ignored (AddJson is marked as optional).